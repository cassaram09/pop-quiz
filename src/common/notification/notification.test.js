import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, render, mount } from 'utils/enzyme';

import Notification from './notification';

describe('Notification', () => {
  it('renders without crashing', () => {
    const notification = shallow(<Notification />);
  
    expect(notification).toMatchSnapshot();
  });

  it('can handle null props', () => {
    const notification = mount(<Notification />);
    
    expect(notification.props().message).toEqual('');
    expect(notification.props().accept).toEqual({});
    expect(notification.props().cancel).toEqual({});
  });

  it('renders a message', () => {
    const notification = shallow(<Notification message={"Reveal answer?"} />);
    const message = notification.find('.notification__message'); 
    
    expect(message.text()).toBe('Reveal answer?');
  });

  it('calls the accept function prop when accepting the notification', () => {
    console.log = jest.fn();

    const notification = shallow(<Notification accept={{ text: 'Accept', fn: () => console.log('accepted!') }} />);

    notification.find('.notification__accept').simulate('click');
    
    expect(console.log.mock.calls[0][0]).toBe('accepted!')
  });

  it('calls the cancel function prop when cancelling the notification', () => {
    console.log = jest.fn();

    const notification = shallow(<Notification cancel={{ text: 'Cancel', fn: () => console.log('cancelled!') }} />);

    notification.find('.notification__cancel').simulate('click');
    
    expect(console.log.mock.calls[0][0]).toBe('cancelled!')
  });

  it('calls the cancel function prop when deleting the notification', () => {
    console.log = jest.fn();

    const notification = shallow(<Notification cancel={{ text: 'Cancel', fn: () => console.log('deleted!') }} />);

    notification.find('.notification__delete').simulate('click');
    
    expect(console.log.mock.calls[0][0]).toBe('deleted!')
  });
});
