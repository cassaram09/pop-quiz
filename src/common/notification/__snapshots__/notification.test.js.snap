// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Notification
    accept={Object {}}
    cancel={Object {}}
    message=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="delete notification__delete"
        />,
        <p
          className="notification__message"
        >
          
        </p>,
        <button
          className="notification__accept"
        />,
        <button
          className="notification__cancel"
        />,
      ],
      "className": "notification is-warning",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "delete notification__delete",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "notification__message",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "notification__accept",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "notification__cancel",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="delete notification__delete"
          />,
          <p
            className="notification__message"
          >
            
          </p>,
          <button
            className="notification__accept"
          />,
          <button
            className="notification__cancel"
          />,
        ],
        "className": "notification is-warning",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "delete notification__delete",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "notification__message",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "notification__accept",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "notification__cancel",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
